src/Protocol/Protocol_1_9.cpp

```diff
@@ -396,6 +396,33 @@ void cProtocol_1_9_0::SendPlayerSpawn(const cPlayer & a_Player)
 
 
 
+void cProtocol_1_9_0::SendStatistics(const cStatManager & a_Manager)
+{
+	ASSERT(m_State == 3);  // In game mode?
+	int count = a_Manager.GetDirtyStatistics(v1_9StatLookup);
+	// If there are no dirty statistics don't send
+	if (!count)
+	{
+		return;
+	}
```

Comment:
If GetDirtyStatistics returns the array mapping statistic to dirtiness, then `count` can be something like `std::count(..., ..., true)` and the actual value-writing part a loop over the array, that does a lookup for each dirty statistic value

Reply:
the thing is not every value that is dirty might be sent by every protocol. We have to check for every protocol if the dirty stats are the ones to be sent

Reply:
Does the client handle us sending it an empty string/ID of -1?

Reply:
I just checked - in any case this is the marker not to send.

and in returning the empty string/-1 the statistic isn't queued for sending